impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/RelationshipBrowser.java

```diff
@@ -50,158 +53,130 @@
+        RelationshipBrowser b = new RelationshipBrowser(iContext, sourceClass, AbstractGraphService.pathWith
+                (path, goToEdge).andFilter(filters).get());
         return new Relationships.Single() {
 
             @Override
             public Relationship entity() {
-                HawkularPipeline<?, Edge> edges = b.source().bothE().has("id", relationshipId).cast(Edge.class);
+                HawkularPipeline<?, Edge> edges = b.source().cast(Edge.class);
                 if (!edges.hasNext()) {
-                    return null;
+                    throw new RelationNotFoundException(sourceClass, FilterApplicator.filters(b.path));
                 }
                 Edge edge = edges.next();
-                return new Relationship(edge.getId().toString(), edge.getLabel(), convert(edge.getVertex(Direction
-                        .OUT)), convert(edge.getVertex(Direction.IN)));
-            }
-
-            @Override
-            public Tenants.ReadRelate tenants() {
-                return b.tenants(EdgeFilter.ID, relationshipId);
-            }
-
-            @Override
-            public Environments.ReadRelate environments() {
-                return (Environments.ReadRelate) b.<EnvironmentsService>getService(EdgeFilter.ID, relationshipId,
-                        Environment.class, EnvironmentsService.class);
-            }
 
-            @Override
-            public Feeds.ReadRelate feeds() {
-                return (Feeds.ReadRelate)b.<FeedsService>getService(EdgeFilter.ID, relationshipId, Feed.class,
-                        FeedsService.class);
-            }
-
-            @Override
-            public MetricTypes.ReadRelate metricTypes() {
-                return (MetricTypes.ReadRelate)b.<MetricTypesService>getService(EdgeFilter.ID, relationshipId,
-                        MetricType.class, MetricTypesService.class);
-            }
-
-            @Override
-            public Metrics.ReadRelate metrics() {
-                return (Metrics.ReadRelate)b.<MetricsService>getService(EdgeFilter.ID, relationshipId, Metrics.class,
-                        MetricsService.class);
-            }
-
-            @Override
-            public Resources.ReadRelate resources() {
-                return (Resources.ReadRelate)b.<ResourcesService>getService(EdgeFilter.ID, relationshipId, Resource
-                                .class, ResourcesService.class);
-            }
-
-            @Override
-            public ResourceTypes.ReadRelate resourceTypes() {
-                return (ResourceTypes.ReadRelate)b.<ResourceTypesService>getService(EdgeFilter.ID, relationshipId,
-                        ResourceType.class,
-                        ResourceTypesService.class);
+                Relationship relationship = new Relationship(edge.getId().toString(), edge.getLabel(), convert(edge
+                         .getVertex(Direction.OUT)), convert(edge.getVertex(Direction.IN)));
```

Comment:
Ah, it is me who misunderstood blueprints' javadoc. I thought "tail" would mean the "end" of the edge. Their class-level javadoc explains it better.